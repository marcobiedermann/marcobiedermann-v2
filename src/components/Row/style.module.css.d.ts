/* eslint-disable import/prefer-default-export */

export const row: string;
