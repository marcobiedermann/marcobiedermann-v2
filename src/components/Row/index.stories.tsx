/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Row from '.';

storiesOf('Row', module).add('default', () => <Row>Content</Row>);
