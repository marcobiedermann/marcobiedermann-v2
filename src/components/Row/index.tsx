import classNames from 'classnames';
import React from 'react';
import Alignment, { AlignmentProps } from '../Alignment';
import styles from './style.module.css';

export interface RowProps extends AlignmentProps {
  className?: string;
}

const Row: React.FC<RowProps> = (props) => {
  const { children, className, ...otherProps } = props;

  return (
    <Alignment className={classNames(className, styles.row)} {...otherProps}>
      {children}
    </Alignment>
  );
};

export default Row;
