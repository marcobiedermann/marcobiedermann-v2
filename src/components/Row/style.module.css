.row {
  display: grid;
  grid-gap: 3em;
  grid-template-columns: repeat(12, 1fr);
}
