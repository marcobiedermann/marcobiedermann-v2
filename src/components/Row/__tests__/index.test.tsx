/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Row from '..';

describe('Row component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Row />);

    expect(asFragment()).toMatchSnapshot();
  });
});
