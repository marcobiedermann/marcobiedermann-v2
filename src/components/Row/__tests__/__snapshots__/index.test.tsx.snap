// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="row"
  />
</DocumentFragment>
`;
